Point-of-No-Return/State.cpp

```diff
@@ -0,0 +1,77 @@
+#include "State.h"
+
+bool State::Search(Vec2& player_pos, IsReverse& is_reverse)
+{
+
+	if (is_reverse.value == false)
```

Comment:
そもそもboolまでwrapする必要があるのか？とは思いますが、もしやるのであれば
```suggestion
	if (is_reverse == IsReverser(false))
```
の方が良いでしょうね。

Reply:
boolのようなprimitive型をwrapするメリットの一つに、下記のようにーバーロードが絡むパターンが考えられますね。

e.g)
```
void Func(bool isReverse);
void Func(bool isHoge);
```
`isReverse` と `isHoge` は目的が全く異なるフラグだが、どちらもbool型なのでオーバーロード不可。

これを
```
struct IsReverse {
    bool value;
};

struct IsHoge {
    bool value;
};

void Func(IsReverse isReverse);
void Func(IsHoge isHoge);
```
とすれば、 `IsReverse` と `IsHoge` は別の型なのでオーバーロード可能。

Reply:
分かりました
チームで話し合って決めます。

Reply:
@chihiro390 
個人的に、wrapしたほうが間違えが少なくていいと思いました。